docs/source/commands/osnadminchannel.md

```diff
@@ -0,0 +1,207 @@
+                                 the OSN
+      --client-key=CLIENT-KEY    Path to file containing PEM-encoded private key
+                                 to use for mutual TLS communication with the
+                                 OSN
+
+Subcommands:
+  channel join --channel-id=CHANNEL-ID --config-block=CONFIG-BLOCK
+    Join an Ordering Service Node (OSN) to a channel. If the channel does not
+    yet exist, it will be created.
+
+  channel list [<flags>]
+    List channel information for an Ordering Service Node (OSN). If the
+    channel-id flag is set, more detailed information will be provided for that
+    channel.
+
+  channel remove --channel-id=CHANNEL-ID
+    Remove an Ordering Service Node (OSN) from a channel.
+```
+
+
+## osnadmin channel join
+```
+usage: osnadmin channel join --channel-id=CHANNEL-ID --config-block=CONFIG-BLOCK
+
+Join an Ordering Service Node (OSN) to a channel. If the channel does not yet
+exist, it will be created.
+
+Flags:
+      --help                     Show context-sensitive help (also try
+                                 --help-long and --help-man).
+  -o, --orderer-address=ORDERER-ADDRESS  
+                                 Endpoint of the OSN
+      --ca-file=CA-FILE          Path to file containing PEM-encoded trusted
+                                 certificate(s) for the OSN
+      --client-cert=CLIENT-CERT  Path to file containing PEM-encoded X509 public
+                                 key to use for mutual TLS communication with
+                                 the OSN
+      --client-key=CLIENT-KEY    Path to file containing PEM-encoded private key
+                                 to use for mutual TLS communication with the
+                                 OSN
+  -c, --channel-id=CHANNEL-ID    Channel ID
+  -b, --config-block=CONFIG-BLOCK  
+                                 Path to the file containing the config block
+```
+
+
+## osnadmin channel list
+```
+usage: osnadmin channel list [<flags>]
+
+List channel information for an Ordering Service Node (OSN). If the channel-id
+flag is set, more detailed information will be provided for that channel.
+
+Flags:
+      --help                     Show context-sensitive help (also try
+                                 --help-long and --help-man).
+  -o, --orderer-address=ORDERER-ADDRESS  
+                                 Endpoint of the OSN
+      --ca-file=CA-FILE          Path to file containing PEM-encoded trusted
+                                 certificate(s) for the OSN
+      --client-cert=CLIENT-CERT  Path to file containing PEM-encoded X509 public
+                                 key to use for mutual TLS communication with
+                                 the OSN
+      --client-key=CLIENT-KEY    Path to file containing PEM-encoded private key
+                                 to use for mutual TLS communication with the
+                                 OSN
```

Comment:
How do they know that these 2 fields are required and not optional throughout these commands?

Reply:
The flags are all set as required in the implementation (the only optional flag for any the commands is --channel-id on the list command). The CLI will return an error if any required flags are not set. Do we do anything to differentiate the required flags for other commands that you're aware of?

Reply:
OK makes sense. Then I would suggest we update the list command so they know that --channel-id is optional -- if not specified they will get back the details of all the channels the osn is a member of instead of a specific one.

Reply:
I have that included in the description for list: "If the channel-id flag is set, more detailed information will be provided for that channel." and also mention it in the example usage for the command. 

Do you have specific places you'd like to see more added?

Reply:
For the --ca-file flag, how would they know this is the root cert of the TLS CA and not the organization CA?
Should it be:
Path to file containing PEM-encoded trusted TLS certificate(s) for the OSN
or more specifically:
Path to file containing PEM-encoded trusted root certificate(s) of the TLS CA for the OSN

Reply:
I reused the flag description that we've been using for years on the `peer channel` flag of the same name. I can certainly update this if we think it's beneficial.

Reply:
Updated to `Path to file containing PEM-encoded CA certificate(s) for the OSN`